Add arm64 centos7

DrewScoggins opened the pull request:
This adds a Arm64 Centos 7 image that matches the one that we use for Linux Amd64 optimization data gathering.

mthalman commented on src/centos/manifest.json:
```diff
@@ -18,6 +18,23 @@
+              "osVersion": "centos7",
+              "tags": {
+                "centos-7-arm64v8-$(System:TimeStamp)-$(System:DockerfileGitCommitSha)": {},
+                "centos-7-arm64v8": {
+                  "isLocal": true
+                }
```
Not needed since there is no dependent Dockerfile that needs to reference this tag.

mthalman reviewed (commented):
This will also require updating the build timeouts to account for arm. Please add 
 the following parameter, `linuxArmBuildJobTimeout: 210`, to https://github.com/dotnet/dotnet-buildtools-prereqs-docker/blob/31c08723eca277df5b0e044b261d438a7578f5c7/eng/pipelines/dotnet-buildtools-prereqs-centos-pr.yml#L24-L27 and https://github.com/dotnet/dotnet-buildtools-prereqs-docker/blob/31c08723eca277df5b0e044b261d438a7578f5c7/eng/pipelines/dotnet-buildtools-prereqs-centos.yml#L24-L27

mthalman commented on eng/common/templates/stages/dotnet/build-test-publish-repo.yml:
```diff
@@ -13,6 +13,7 @@ parameters:
   windowsAmdBuildJobTimeout: 60
   windowsAmdTestJobTimeout: 60
   linuxAmdBuildJobTimeout: 60
+  linuxAmdTestJobTimeout: 60
```
The changes in this file are not needed.

mthalman commented on eng/pipelines/dotnet-buildtools-prereqs-centos.yml:
```diff
@@ -25,3 +25,4 @@ stages:
     internalProjectName: ${{ variables.internalProjectName }}
     publicProjectName: ${{ variables.publicProjectName }}
     linuxAmdBuildJobTimeout: 210
+    linuxAmdTestJobTimeout: 210
```
```suggestion
    linuxArmBuildJobTimeout: 210
```

The same parameter needs to be added to: https://github.com/dotnet/dotnet-buildtools-prereqs-docker/blob/31c08723eca277df5b0e044b261d438a7578f5c7/eng/pipelines/dotnet-buildtools-prereqs-centos-pr.yml#L24-L27

DrewScoggins replied:
OK, I was wondering where I needed to add that. I will make these changes.

DrewScoggins commented:
I believe I have now addressed all of the feedback. I am going to make sure that the build of the new image completes, and then hopefully we should be good to go.

DrewScoggins commented:
I don't have write access in the this repo. So when it finishes, I will likely ping you to hit the merge button :)

DrewScoggins commented:
These changes also timeout, I am going to go back to the previous changes, as they at least allowed the build to complete.

mthalman commented on eng/pipelines/dotnet-buildtools-prereqs-centos.yml:
```diff
@@ -25,3 +25,4 @@ stages:
     internalProjectName: ${{ variables.internalProjectName }}
     publicProjectName: ${{ variables.publicProjectName }}
     linuxAmdBuildJobTimeout: 210
+    linuxArmBuildJobTimeout: 210
```
Looks like the timeout needs to be increased.
```suggestion
    linuxArmBuildJobTimeout: 270
```

mthalman replied:
This same timeout parameter needs to be added here:

https://github.com/dotnet/dotnet-buildtools-prereqs-docker/blob/26b5ef43b338cd9f130ec4ee03994886947f8896/eng/pipelines/dotnet-buildtools-prereqs-eng.yml#L25-L30

mthalman commented:
@DrewScoggins - This change still needs to be made: https://github.com/dotnet/dotnet-buildtools-prereqs-docker/pull/653#discussion_r940202043

DrewScoggins commented:
Yeah, I just made the change.

DrewScoggins commented:
@mthalman Looks like this has passed. I am going to up the timeout to 300 minutes however, as the current runs take about 276 of the 280 allotted minutes. But once I make the change, could you click the merge button?
